drv/lpc55-update-server/src/main.rs

```diff
@@ -529,6 +574,23 @@ impl idl::InOrderUpdateImpl for ServerImpl<'_> {
             }
         }
     }
+
+    fn component_switch_cancel_pending(
+        &mut self,
+        _: &userlib::RecvMessage,
+        component: RotComponent,
+        slot: SlotId,
+        duration: SwitchDuration,
+    ) -> Result<(), RequestError<UpdateError>> {
+        match component {
+            RotComponent::Hubris => {
+                self.switch_default_hubris_image_cancel_pending(slot, duration)
+            }
+            RotComponent::Stage0 => {
+                Err(UpdateError::InvalidSlotIdForOperation.into())
+            }
```

Comment:
Hmmm, It might be cleaner to just error out in `mgs_common.rs` because the issue is the component, not the Slot ID.